pool.go

```diff
@@ -0,0 +1,192 @@
+package server
+
+import (
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/bblfsh/sdk/protocol"
+	"github.com/pkg/errors"
+	"golang.org/x/time/rate"
+)
+
+const (
+	DefaultPoolMin               = 1
+	DefaultPoolMax               = 5
+	DefaultPoolTimeBeforeNew     = time.Second * 3
+	DefaultPoolTimeBetweenSpawns = time.Second * 10
+	DefaultPoolTimeBeforeClean   = time.Minute * 1
+)
+
+// DriverPool controls a pool of drivers and balances requests among them,
+// ensuring each driver does not get concurrent requests.
```

Comment:
It is important to explain that drivers are reused after they process a request as there are pools that don't reuse their resources.